src/compiler/checker.ts

```diff
@@ -570,6 +555,8 @@ namespace ts {
             getFullyQualifiedName,
             getResolvedSignature: (node, candidatesOutArray, argumentCount) =>
                 getResolvedSignatureWorker(node, candidatesOutArray, argumentCount, CheckMode.Normal),
+            getResolvedSignatureForStringLiteralCompletions: (call, editingArgument, candidatesOutArray) =>

```

Comment:
Can't we just add an `@internal` overload for `getResolvedSignature` that allows `CheckMode` to be passed in?

Reply:
`editingArgument` is also needed, and `argumentCount` isn’t needed... and I like that the name implies the signature you’re going to get is in fact not at all the same signature you would get for normal checking purposes.